extensions/workouts/src/calculator.tsx

```diff
@@ -0,0 +1,146 @@
+  const [unit, setUnit] = useCachedState<"km" | "mi">("unit", "km");
+  const [selectedPreset, setSelectedPreset] = useState<string>("");
+
+  useEffect(() => {
+    const values = {
+      distance,
+      time,
+      pace,
+      distanceUnit: unit,
+      mode,
+    };
+    if (mode === "pace" && distance && time) {
+      const calculatedPace = calculatePace(values);
+      setCalculatedPace(calculatedPace);
+    } else if (mode === "time" && distance && pace) {
+      const calculatedTime = calculateTime(values);
+      setCalculatedTime(calculatedTime);
+    }
+  }, [distance, time, pace, unit, mode]);
+
+  const handlePresetChange = (preset: string) => {
+    setSelectedPreset(preset);
+    const isPreset = (p: string): p is DistancePreset => p in distancePresets;
+
+    if (isPreset(preset)) {
+      setDistance(distancePresets[preset][unit]);
+    } else {
+      setDistance("");
+    }
+  };
+
+  const calculatePace = (values: PaceCalculatorForm) => {
+    const distance = parseFloat(values.distance.replace(",", ".")) || 0;
+    if (distance === 0) return "00:00";
+
+    // Parse time (HH:MM:SS format)
+    const [hours, minutes, seconds] = values.time.split(":").map(Number);
+    const totalMinutes = hours * 60 + (minutes || 0) + (seconds || 0) / 60;
+
+    // Calculate pace (minutes per unit)
+    const paceMinutes = totalMinutes / distance;
+
+    // Convert to HH:MM:SS format
+    const paceHours = Math.floor(paceMinutes / 60);
+    const paceRemainingMinutes = Math.floor(paceMinutes % 60);
+    const paceSeconds = Math.floor((paceMinutes * 60) % 60);
+
+    return `${String(paceHours).padStart(2, "0")}:${String(paceRemainingMinutes).padStart(2, "0")}:${String(paceSeconds).padStart(2, "0")}`;
+  };
+
+  const calculateTime = (values: PaceCalculatorForm) => {
+    const distance = parseFloat(values.distance.replace(",", ".")) || 0;
+    if (distance === 0) return "00:00:00";
+
+    // Parse pace (MM:SS format)
+    const [paceMinutes, paceSeconds] = values.pace.split(":").map(Number);
+    const paceInMinutes = (paceMinutes || 0) + (paceSeconds || 0) / 60;
+
+    // Calculate total time
+    const totalMinutes = distance * paceInMinutes;
+
+    // Convert to HH:MM:SS format
+    const resultHours = Math.floor(totalMinutes / 60);
+    const resultMinutes = Math.floor(totalMinutes % 60);
+    const resultSeconds = Math.floor((totalMinutes * 60) % 60);
+
+    return `${String(resultHours).padStart(2, "0")}:${String(resultMinutes).padStart(2, "0")}:${String(resultSeconds).padStart(2, "0")}`;
+  };
+
+  return (
+    <Form>
+      <Form.Dropdown
+        id="mode"
+        title="Calculation Mode"
+        value={mode}
+        onChange={(newValue) => setMode(newValue as CalculationMode)}
+      >
+        <Form.Dropdown.Item value="pace" title="Calculate Pace" />
+        <Form.Dropdown.Item value="time" title="Calculate Time from Pace" />
+      </Form.Dropdown>
+
+      <Form.Dropdown
```

Comment:
I think this field + distance could be one combobox that suggests the presets as default values